backend/remote-state/kubernetes/backend.go

```diff
@@ -0,0 +1,359 @@
+package kubernetes
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/hashicorp/terraform/backend"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/version"
+	"github.com/mitchellh/cli"
+	"github.com/mitchellh/go-homedir"
+	k8sSchema "k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+)
+
+// Modified from github.com/terraform-providers/terraform-provider-kubernetes
+
+const (
+	noConfigError = `
+
+[Kubernetes backend] Neither service_account nor load_config_file were set to true, 
+this could cause issues connecting to your Kubernetes cluster.
+`
+)
+
+var (
+	secretResource = k8sSchema.GroupVersionResource{
+		Group:    "",
+		Version:  "v1",
+		Resource: "secrets",
+	}
+)
+
+// New creates a new backend for kubernetes remote state.
+func New() backend.Backend {
+	s := &schema.Backend{
+		Schema: map[string]*schema.Schema{
+			"key": {
```

Comment:
I'm not sure about the naming here. This is called `key` with the ambiguous description below - but what this actually does is adds a label to the Secret called `terraKey` and suffixes this to the the Secret name.

Reply:
Perhaps we should be explicit about this like we are in other backends, and have `secret_name` attribute that is the exact name of the secret where we are storing the state. i.e S3 has `bucket` and `key`, pg has `schema_name`. 

We could have an additional `labels` attribute if we wanted to supply additional labels.

Reply:
I see that the name and labels are being used to differentiate workspace, so maybe `secret_suffix` would make sense here.